controllers/territory_controller/territory_controller.py

```
@@ -150,9 +150,11 @@ def claim_territory(
             return
 
         new_colour = ZONE_COLOURS[claimed_by]
-        self._robot.getFromDef(station_code).getField("zoneColour").setSFColor(
-            list(new_colour),
-        )
+        station = self._robot.getFromDef(station_code)
```

When's this `None`?

---

In theory never but since the type is `Optional[Node]` we have to handle it

---

Presumably it's `None` if the name isn't found?

---

That is true, we should probably throw an error if we can't find a territory.